### PR TITLE
Add user verification and resident key support

### PR DESCRIPTION
**Pull request by syntaxcase**

### CODE REVIEW

File: fido2/server.py [hunk 1]
```diff
@@ -49,13 +49,33 @@ class ATTESTATION(six.text_type, Enum):
...
-    def __init__(self, rp, attestation=None, verify_origin=None):
+    def __init__(
+            self,
+            rp,
+            attestation=None,
+            verify_origin=None,
+            resident_key=False,
+            user_verification=USER_VERIFICATION.PREFERRED
+    ):
+        if user_verification not in USER_VERIFICATION:
```
Comment (dainnilsson) [inline fido2/server.py]: I'd change this to make it possible to pass in a string as well. Now, for example:
`'discouraged' in USER_VERIFICATION == False`.

Maybe change it to: `self.user_verification = USER_VERIFICATION(user_verification)`?
That has the benefit of already throwing a ValueError if something weird is passed.

File: fido2/server.py [hunk 2]
```diff
@@ -49,13 +49,33 @@ class ATTESTATION(six.text_type, Enum):
...
+            raise ValueError('"{}" is not a member of {}'
+                             .format(user_verification, USER_VERIFICATION))
+
         self.rp = rp
         self._verify = verify_origin or _verify_origin_for_rp(rp['id'])
         self.timeout = 30
         self.attestation = attestation or ATTESTATION.NONE
```
Comment (dainnilsson) [inline fido2/server.py]: Come to think of it, we should probably do `self.attestation = ATTESTATION(attestation)` here as well, and change the default from `None` to `ATTESTATION.NONE`. Would you agree?

Comment (syntaxcase) [reply fido2/server.py]: Agreed.

File: fido2/server.py (hunk 1 above)
Comment (syntaxcase) [reply fido2/server.py]: I agree, no need to import the enum, which is nice.

**Pull request merged**